application/ui/components/param-row.jsx

```diff
@@ -0,0 +1,221 @@
+/* global console */
+'use strict';
+
+var _               = require('underscore');
+var React           = require('react');
+var marked          = require('marked');
+var Text            = require('./param/text');
+var Select          = require('./input/select');
+var StripeToken     = require('./input/stripe-token');
+var CustomObject    = require('./input/custom-object');
+var ArrayObject     = require('./input/array');
+var Params;
+
+module.exports = React.createClass({
+    displayName : 'Param',
+
+    propTypes : {
+        param     : React.PropTypes.object,
+        onChange  : React.PropTypes.func,
+        onInclude : React.PropTypes.func,
+        onNull    : React.PropTypes.func,
+        simple    : React.PropTypes.bool
+    },
+
+    validateParamConfiguration : function(param)
+    {
+        var errors = [];
+
+        switch (param.type) {
+            case "enum":
```

Comment:
Since this switch block only has one case, we could replace that with `if (param.type === 'enum')` and get rid of a nesting leve.